content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,859 @@
+
+### ANOVA with Python implementation
+
+ANOVA is a word coined from 'Analysis of Variance'. It is a statistical concept that shows the differences between the means of more than two independent groups, using variance analysis on samples from those groups.
+
+
+It is used for checking the contrast between three or more samples with one test. Especially when the categorical class has over two categories.
+
+Why ANOVA? With ANOVA, you get to discover obvious differences between the means of your independent features, and on getting a clearer picture of the differences, you can get to understand how each of them connects to your dependent variable, and what are the influencing factors for the relationship.
+
+A typical example is as follows:
+
+
+'The growth department of a certain company discovered an increase in sales for a particular product. In recent times, they've tried several advertisement channels so it is difficult to say which one yielded the sales increase the most.
+
+They did online posters, billboards, TV ads, mouth-to-mouth, and sponsorship. By carrying out an ANOVA test on the several advertisement channels, they will be able to better understand which one works best out of all channels(independent variables), with the connection it has with the increase in sales which is the dependent variable.
+
+
+This can be applied in the marketing department of every organization.
+
+During ANOVA testing, the hypothesis is:
+
+**H0:** When all samples' means are the same.
+
+**H1:** When one or more samples are very much different.
+
+#### One Way ANOVA Test
+
+`petal width`, `petal length` and `sepal length` are the classes we want to apply one-way ANOVA test on.
+
+```python
+
+class1 = df.petal_width
+
+class2 = df.sepal_length
+
+class3 = df.petal_length
+
+
+print(f_oneway(class1, class2, class3))
+
+```
+
+```bash
+
+F_onewayResult(statistic=555.8702565323317, pvalue=5.738282463819433e-122)
+
+```
+
+Since our `p-value` (5.738282463819433e-122) is far less than `alpha` = 0.05, we dismiss the null hypothesis, as there exists no evidence sustainable enough to accept it.
+
+This means that the sample means are very much different. Hence, our H1 (alternate hypothesis) is thus true.
+
+#### Two Way ANOVA Test
+
+This is called for when we are dealing with three or more variables, trying to compare their means with each other.
+
+
+```python
+
+petal = df.petal_width
+
+sepal = df.sepal_length
+
+species = df.species
+
+model = ols('species ~ C(petal) + C(sepal) + C(petal):C(sepal)', data=df).fit()
+
+print(sm.stats.anova_lm(model, typ=2))
+
+```
+
+![twowayanova](/engineering-education/statistical-hypothesis-testing-python-implementation/twowayanova.PNG)
+
+The p-value of `petal width` is more than 0.05, which increases the probability of our null hypothesis being accepted.
```

Comment:
I believe it is `petal_width`. Make all such changes for similar instances.
The edit applies to the sentence below this as well.